src/libgap-api.h

```diff
@@ -3,9 +3,93 @@
 #ifndef LIBGAP_API_H
 #define LIBGAP_API_H
 
+#include "gasman.h"
 #include "system.h"
 
 
+#ifdef __GNUC__
+#define unlikely(x)     __builtin_expect(!!(x), 0)
+#else
+#define unlikely(x)     (x)
+#endif
+
+
+#ifndef GAP_ENTER_DEBUG
+#define GAP_ENTER_DEBUG 0
+#endif
+
+
+extern syJmp_buf * _GAP_GetReadJmpError(void);
+extern Int  _GAP_GetEnterStackCount(void);
+extern void _GAP_IncEnterStackCount(void);
+extern void _GAP_DecEnterStackCount(void);
+extern void _GAP_SetEnterStackCount(Int count);
+
+
+#if GAP_ENTER_DEBUG
+#define GAP_ENTER_DEBUG_MESSAGE(message, file, line) \
+    fprintf(stderr, "%s: %d; %s:%d\n", message, _GAP_EnterStackCount, file, line);
+#else
+#define GAP_ENTER_DEBUG_MESSAGE(message, file, line)
+#endif
+
+
+#define GAP_EnterStack() \
+    GAP_ENTER_DEBUG_MESSAGE("EnterStack", __FILE__, __LINE__); \
+    Int _gap_tmp_enter_stack_count = _GAP_GetEnterStackCount(); \
+    if (_gap_tmp_enter_stack_count < 0) { \
+        _GAP_SetEnterStackCount(-_gap_tmp_enter_stack_count); \
+    } else { \
+        if (_gap_tmp_enter_stack_count == 0) { \
+            MarkStackBottomBags(); \
+        } \
+        _GAP_IncEnterStackCount(); \
+    }
+
+
+#define GAP_LeaveStack() \
+    _GAP_DecEnterStackCount(); \
+    GAP_ENTER_DEBUG_MESSAGE("LeaveStack", __FILE__, __LINE__);
+
+
+static inline int _GAP_Error_Prejmp(const char* file, int line) {
```

Comment:
Is there a formatting guide for GAP somewhere?  In other commits I've tried to follow the style just based on what I saw in the file; here I wasn't worrying about it as much because it was more a prototype anyways.

Reply:
Broadly speaking, the format is use clang-format, except there is lots of old code with lots of different styles, which we have decided not to reformat on bulk as it would "break history", making things like `git blame` useless, mess up forks, etc.